src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -233,31 +233,77 @@ private async Task ProcessBuildNotificationsAsync(Build build)
+                    bool updateExisting = repo.UpdateExisting;
 
-                    var newIssue =
-                        new NewIssue($"Build failed: {build.Definition.Name}/{prettyBranch} #{build.BuildNumber}")
-                        {
-                            Body = body,
+                    if (updateExisting)
+                    {
+                        RepositoryIssueRequest issueRequest = new RepositoryIssueRequest {
+                            Creator = "dotnet-eng-status[bot]",
+                            State = ItemStateFilter.Open,
+                            SortProperty = IssueSort.Created,
+                            SortDirection = SortDirection.Descending
                         };
 
-                    if (!string.IsNullOrEmpty(monitor.Assignee))
-                    {
-                        newIssue.Assignees.Add(monitor.Assignee);
-                    }
-                    
-                    foreach (string label in repo.Labels.OrEmpty())
-                    {
-                        newIssue.Labels.Add(label);
+                        if (!string.IsNullOrEmpty(monitor.Assignee))
+                        {
+                            issueRequest.Assignee = monitor.Assignee;
+                        }
+
+                        foreach (string label in repo.Labels.OrEmpty())
+                        {
+                            issueRequest.Labels.Add(label);
+                        }
+
+                        foreach (string label in monitor.Labels.OrEmpty())
+                        {
+                            issueRequest.Labels.Add(label);
+                        }
+
+                        List<Issue> matchingIssues = (await github.Issue.GetAllForRepository(repo.Owner, repo.Name, issueRequest)).ToList();
+                        Issue matchingIssue = matchingIssues.FirstOrDefault(i => i.Title.StartsWith(issueTitlePrefix));
+
+                        if (matchingIssue != null)
+                        {
+                            // Add a new comment to the issue with the body
+                            IssueComment newComment = await github.Issue.Comment.Create(repo.Owner, repo.Name, matchingIssue.Id, body);
+                            _logger.LogInformation("Logged comment in {owner}/{repo}#{issueNumber} for build failure", repo.Owner, repo.Name, matchingIssue.Number);
+                        }
+                        else
+                        {
+                            _logger.LogInformation("Matching issues for {issueTitlePrefix} not found. Creating a new issue.", issueTitlePrefix);
+                            updateExisting = false;
+                        }
                     }
 
-                    foreach (string label in monitor.Labels.OrEmpty())
+                    // Create new issue if repo.UpdateExisting is false or there were no matching issues
+                    if (!updateExisting)
```

Comment:
Line 275 turns it to false if we couldn't find the issue.

I'd probably refactor this a bit and make that an early return, so that we don't need this boolean floating around, something like:
```
if (updateExisting) {
  // stuff
  if (found) {
    // do update
    return
}

// make new issue
```

Reply:
(Since the modifyable boolean can easily be missed.  I missed it the first time reading through the code too, and had to do a bit of a double take to find other the assignment.

Reply:
I knew there was a better way to do it! Thanks guys :)